runtime/src/enclave_rpc/client.rs

```diff
@@ -456,48 +108,103 @@ impl<T> Response<T> {
     }
 
     /// Send peer feedback.
-    async fn send_peer_feedback(&mut self, pf: types::PeerFeedback) {
+    async fn send_peer_feedback(&mut self, feedback: types::PeerFeedback) {
         if let Some(request_id) = self.request_id.take() {
             // Only count feedback once.
-            if let Some(cmdq) = self.cmdq.upgrade() {
-                let _ = cmdq
-                    .send(Command::PeerFeedback(request_id, pf, self.kind))
-                    .await;
-            }
+            let _ = self
+                .transport
+                .submit_peer_feedback(request_id, feedback)
+                .await; // Ignore error.
         }
     }
 }
 
 /// RPC client.
 pub struct RpcClient {
-    /// Internal command queue (sender part).
-    cmdq: mpsc::Sender<Command>,
+    /// Used transport.
+    transport: Box<dyn Transport>,
+    /// Multiplexed sessions.
+    sessions: tokio::sync::Mutex<Sessions<signature::PublicKey>>,
+    /// The ID of the client.
+    client_id: u32,
+    /// The ID of the next transport request.
+    next_request_id: AtomicU32,
 }
 
 impl RpcClient {
-    fn new(transport: Box<dyn Transport>, builder: Builder) -> Self {
-        // Create the command channel.
-        let (tx, rx) = mpsc::channel(CMDQ_BACKLOG);
-
-        // Ensure every client has a unique ID.
+    fn new(
+        transport: Box<dyn Transport>,
+        builder: Builder,
+        max_sessions: usize,
+        max_sessions_per_peer: usize,
+        stale_session_timeout: i64,
+    ) -> Self {
+        // Assign a unique ID to each client to avoid overlapping request IDs.
         let client_id = NEXT_CLIENT_ID.fetch_add(1, Ordering::SeqCst); // Wraps if overflows.
+        let next_request_id = AtomicU32::new(1);
 
-        // Create the controller task and start it.
-        let controller = Controller {
-            session: MultiplexedSession::new(builder),
+        let sessions = tokio::sync::Mutex::new(Sessions::new(
+            builder,
+            max_sessions,
+            max_sessions_per_peer,
+            stale_session_timeout,
+        ));
+
+        Self {
             transport,
-            cmdq: rx,
+            sessions,
             client_id,
-            sent_request_count: 0,
+            next_request_id,
+        }
+    }
+
+    /// Construct an unconnected RPC client with runtime-internal transport.
+    pub fn new_runtime(
+        protocol: Arc<Protocol>,
+        endpoint: &str,
+        builder: Builder,
+        max_sessions: usize,
+        max_sessions_per_peer: usize,
+        stale_session_timeout: i64,
+    ) -> Self {
+        let transport = Box::new(RuntimeTransport::new(protocol, endpoint));
+
+        Self::new(
+            transport,
+            builder,
+            max_sessions,
+            max_sessions_per_peer,
+            stale_session_timeout,
+        )
+    }
+
+    /// Update allowed remote enclave identities.
+    ///
+    /// Useful if the key manager's policy has changed.
```

Comment:
Not sure if we should mention the key manager in the generic EnclaveRPC client?

Reply:
Definitely not. Removing as this was copy pasted from the key manager client.